<!doctype html>
<html>
<head>
	<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
	<meta charset="utf-8">
	
		<title>No success or error callbacks after registering for push notifications on iOS: Titanium SDK 3.0 » Community Questions &amp; Answers </title>
	
	<meta name="viewport" content="width=device-width,initial-scale=1">
	<link href="../../related/favicon.png" rel="shortcut icon" type="image/png">
	<link href="//fonts.googleapis.com/css?family=Open+Sans:300,400,500,600,700" rel="stylesheet" type="text/css">
	<link href="//d3ilu1xuwhtfe2.cloudfront.net/e8e0ebd/css/style.css" rel="stylesheet" type="text/css">
	<link href="../../related/qa.css" rel="stylesheet" type="text/css">
	<link href="../../related/hybrid.css" rel="stylesheet" type="text/css">
	
	<link href="../../related/overrides.css" rel="stylesheet" type="text/css">
</head>
<body>

<div class='mast'>    <div class='container'>        <div class='row'>            <div class='col-sm-12'>                <div class='masthead'><h1>Titanium Community Questions & Answer Archive</h1><h2>We felt that 6+ years of knowledge should not die so this is the Titanium Community Questions &amp; Answer Archive</h2></div>            </div>        </div>    </div></div>

<main class="container">
	<div class="row">
		<div class="col-sm-12">
			
		</div>
	</div>
	<div class="row">
		<div class="col-sm-12">
			<article class="question-container">
	<div class="question">
		<header>
			<h1>No success or error callbacks after registering for push notifications on iOS: Titanium SDK 3.0</h1>
		</header>
		<section>
			<p>Titanium 3.0 SDK.<br>Tested on iOS 6.0 and 5.1 on physical devices.</p>
<p>When calling Ti.Network.registerForPushNotifications() I&#39;m successfully seeing the Allow&#x2F;Deny alert prompt however when clicking Allow I see no Success or Error callbacks fired. This of course means I can&#39;t obtain the deviceToken.</p>
<p>I realize this doesn&#39;t work in the Simulator, I&#39;m testing on a physical iPad 2 and iPad 3. I also realize the Push SSL Certificates need to be generated and added to my App ID within the Provisioning Portal (and then obviously uploaded to your APNs provider e.g. Urban Airship). And that the App ID cannot contain the * wildcard for Push to work. I&#39;ve done all this.</p>
<p>Clicking Allow successfully adds the app under the Notifications section of Settings of course without the deviceToken that&#39;s useless.</p>
<p>I&#39;m stumped why my success or error callbacks aren&#39;t firing? Below is my code:</p>
<pre><code class="hljs">Ti.Network.registerForPushNotifications({
    types: [Ti.Network.NOTIFICATION_TYPE_BADGE,
        Ti.Network.NOTIFICATION_TYPE_ALERT,
        Ti.Network.NOTIFICATION_TYPE_SOUND],
    success: <span class="hljs-function"><span class="hljs-keyword">function</span><span class="hljs-params">(e)</span> </span>{
        Ti.API.info(<span class="hljs-string">"PUSH SUCCESS: token = "</span>+e.deviceToken);
        <span class="hljs-keyword">var</span> deviceToken = e.deviceToken;
    },
    error: <span class="hljs-function"><span class="hljs-keyword">function</span><span class="hljs-params">(e)</span> </span>{
        Ti.API.info(<span class="hljs-string">"PUSH FAILED"</span>);
    },
    <span class="hljs-keyword">callback</span>: <span class="hljs-function"><span class="hljs-keyword">function</span><span class="hljs-params">(e)</span> </span>{
        <span class="hljs-comment">// Ignoring for now...</span>
    }
});
</code></pre><p>Furthermore considering I haven&#39;t received the deviceToken, what&#39;s the best way to reset the app&#x2F;device to re-prompt for Push Notifications permission again (during testing)? Sadly it&#39;s not as easy as changing the app&#39;s version number, App ID (due to invalidating your push certificate), nor calling <code>Ti.Network.unregisterForPushNotifications();</code> then re-calling <code>Ti.Network.registerForPushNotifications({...})</code>. Also uninstalling and waiting 24 hours as some have suggested didn&#39;t seem to reset the saved permissions either.</p>
<p>Cheers,</p>
<p>Tim</p>

		</section>
		<footer>
			<div class="author">
				&mdash; asked <span title="February 6th 2013, 10:24:59 pm">February 6th 2013</span>
				by <span class='authorname'>Tim Keir</span>
			</div>

			
				<ul class="tags">
					
						<li><span class='tagname'>alloy</span></li>
					
						<li><span class='tagname'>apns</span></li>
					
						<li><span class='tagname'>ios</span></li>
					
						<li><span class='tagname'>notification</span></li>
					
						<li><span class='tagname'>push</span></li>
					
				</ul>
			

			<section>
				<h5>1 Comment</h5>
				
					<ul class="comments">
					
						<li class="comment">
							<div><p>bump</p>
</div>
							<div class="author">
								&mdash; commented <span title="March 4th 2013, 12:04:32 am">March 4th 2013</span>
								by <span class='authorname'>Tim Keir</span>
							</div>
						</li>
					
					</ul>
				
			</section>
		</footer>
		<aside class="vote-box">
			<div class="score"><span>0</span> Votes</div>
			<div class="answers"><span>2</span> Answers</div>
		</aside>
	</div>

	<hr>

	<h3>2 Answers</h3>

	
		<ul class="answers">
			
				<li class="answer">
				
					<article id="answer-260260">
				
						<section>
							<p>It turned out this was because the required ports are closed behind our Firewall. Specifically port 5223. More info here:<br>http:&#x2F;&#x2F;support.apple.com&#x2F;kb&#x2F;TS4264?viewlocale=en_US</p>
<p>Because these were WiFi only iPads they didn&#39;t have a cellular data connection to bypass such issues. iOS opts to use the GSM connection to register for push (even if connected to WiFi) to prevent such issues.</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="April 8th 2013, 5:25:56 am">April 8th 2013</span>
								by <span class='authorname'>Tim Keir</span><br>
								<a class="icon-bg icon-link" href="../../question/147956/no-success-or-error-callbacks-after-registering-for-push-notifications-on-ios-titanium-sdk-30.html#answer-260260" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
				<li class="answer">
				
					<article id="answer-287481">
				
						<section>
							<p>I am having the same issue… nothing seems to allow registerForPushNotification to return anything of use or a device &#x2F; token.  Very upsetting for an app that was already working..</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="May 25th 2015, 6:40:11 pm">May 25th 2015</span>
								by <span class='authorname'>Chad Stachowicz</span><br>
								<a class="icon-bg icon-link" href="../../question/147956/no-success-or-error-callbacks-after-registering-for-push-notifications-on-ios-titanium-sdk-30.html#answer-287481" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
		</ul>
	
</article>

		</div>
		<div class="col-sm-3"></div>
	</div>
	<div class="row"><div class="signoff">The ownership of individual contributions to this community generated content is retained by the authors of their contributions.<br>All trademarks remain the property of the respective owner.</div></div>
</main>


</body>
</html>
